src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -314,29 +329,41 @@ protected override void ApplyParseOptionsChanged(ProjectId projectId, ParseOptio
                 throw new ArgumentNullException(nameof(options));
             }
 
-            var parseOptionsService = CurrentSolution.GetProject(projectId).LanguageServices.GetService<IParseOptionsService>();
-            Contract.ThrowIfNull(parseOptionsService, nameof(parseOptionsService));
+            var storage = ProjectPropertyStorage.Create(TryGetDTEProject(projectId), DeferredState.ServiceProvider);
 
-            string newVersion = parseOptionsService.GetLanguageVersion(options);
+            switch (options)
+            {
+                case CSharpCompilationOptions csharpOptions:
+                    storage.SetProperty("AllowUnsafeBlocks", nameof(ProjectConfigurationProperties3.AllowUnsafeBlocks),
+                        csharpOptions.AllowUnsafe.ToString().ToLowerInvariant());
```

Comment:
why the "ToLowerInvariant"?

Reply:
because bool is lower case true/false in xml

Reply:
Looking at our own .csproj files though, i see things like this:

```xml
      <AutoGen>True</AutoGen>
      <DesignTime>True</DesignTime>
```

Reply:
Looks like both are supported, and we also use a mix of both ourselves.  As such, since msbuild just sits on .net, i would just keep the values as is.  Can you even just pass the bool value without changing to a string?  or does SetProperty require a string?

Reply:
Visual Studio generates lowercase, so I made it lowercase

Reply:
Can you just pass the bool along unadulterated, allowing the underlying system to decide what to do with it?  i.e. make your SetProperty take an object, and just pass that to either of the two systems?

Reply:
yes it requires a string

Reply:
if I set it the other way through per configuration properties, it is generated lowercase, so i would prefer to keep it lowercase here as well, thanks

Reply:
Understood.  If you can't pass along an actual bool value, and you do have to generate the string yourself, then i'm fine with the behavior here.  Thanks!